src/writeFile.js

```diff
@@ -22,7 +26,42 @@ export default function writeFile(globalRef, pattern, file) {
         return pify(fs.readFile)(file.absoluteFrom)
         .then((content) => {
             if (pattern.transform) {
-                content = pattern.transform(content, file.absoluteFrom);
+                const transform = (content, absoluteFrom) => {
+                    return pattern.transform(content, absoluteFrom);
+                };
+
+                if (pattern.cache) {
+                    if (!globalRef.cacheDir) {
+                        globalRef.cacheDir = findCacheDir({ name: 'copy-webpack-plugin' });
+                    }
+
+                    const cacheKey = pattern.cacheKey
```

Comment:
@michael-ciniawsky we need another opinion :smile:

Reply:
@michael-ciniawsky btw we can remove this option, because we use `pattern` as part of cache key and if you want invalidate cache you can use `cache` option (example `cache: '/path/to/' + crypto.createHash('md5').update(JSON.stringify(something) || serialize(something)).digest("hex")`)

Reply:
@evilebottnawi If you add it to the `cache` pattern option itself `{ cache: { key: '...' } } || { cache: key }` I'm ok with it, but adding to many (related) options 'top scope' of a pattern is unnecessary (bloat) and can be confusing for folks new to `copy-webpack-plugin` e.g if they assume `pattern.cacheKey` is mandatory to use for some reason (many don't fully read the docs). There are already too many options for patterns and most folks will tend to use simple patterns like `{ from: '', to: '' } || { from: '', to: '', cache: true }` which is sufficient && straight forward for most of the plugin use cases.

Reply:
@michael-ciniawsky btw, sometimes i have external variables for function inside in `transofrm` and i can't invalidate cache without this option, i can only enable or disable cache, but this not convenient. Example i use `nunjucks` inside `transform`, some template render long time and i want to cache this operation, also i know when i should invalidate cache and cacheKey solve this problem.

Reply:
Yep, but this already goes under 'advanced' usage in the scope of `copy-webpack-plugin` imo and wouldn't this work with `{ cache: ${customCacheKey}` instead of `{ cache: true, cacheKey: `${customCacheKey}` }` for some reason ?